### PR TITLE
present_mode, backends and windows size

### PR DESCRIPTION
**Pull request by CatCode79**

### CODE REVIEW

**Comment by CatCode79** [comment]
I also wanted to add the power preference but I discovered that wgpu has a fallback system on the integrated if there is no suitable discrete and vice versa. So it's even less useful than I thought to make it available.

File: src/main.rs [hunk 1]
```diff
@@ -54,17 +54,31 @@ use transform::*;
...
-        .with_title(title)
-        // .with_visible(false)
-        .build(&event_loop)
-        .expect("Failed to create window");
+    let window = {
+        let (width, height) = match event_loop.primary_monitor() {
+            None => (1920, 1080), // Most widespread resolution in 2022.
+            Some(handle) => (handle.size().width, handle.size().height),
+        };
+        let inner_size = winit::dpi::LogicalSize::new(width /2, height /2);
```
Comment (Davidster) [inline src/main.rs:62]: on my computer the logical size causes the window to be quite small. To reproduce the old behavior I had to change this to:

```suggestion
        let inner_size = winit::dpi::PhysicalSize::new(width * 3 / 4, height * 3 / 4);
```

Does that work well on your computer?

File: src/main.rs [hunk 2]
```diff
@@ -54,17 +54,31 @@ use transform::*;
...
+    };
 
-    let base_render_state = BaseRendererState::new(&window).await;
+    let base_render_state = {
+        let backends = if cfg!(target_os = "linux") {
+            wgpu::Backends::from(wgpu::Backend::Vulkan)
+        } else {
+            wgpu::Backends::all()
+        };
+        BaseRendererState::new(&window, backends, wgpu::PresentMode::AutoVsync).await
```
Comment (Davidster) [inline src/main.rs:80]: :+1:

File: src/main.rs (hunk 1 above)
Comment (CatCode79) [reply src/main.rs]: Works without problems!

**Review by Davidster — APPROVED** [review]

**Pull request merged**